service/controller/v4/resource/chart/desired.go

```diff
@@ -20,10 +21,26 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	name := key.ChartName(customObject)
 	channel := key.ChannelName(customObject)
+
+	// Values configmap contains settings managed by the controlling operator.
 	chartConfigmapValues, err := r.getConfigMapValues(ctx, customObject)
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
+
+	// User configmap contains settings overridden by the user.
+	userConfigmapValues, err := r.getUserConfigMapValues(ctx, customObject)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// Merge configmap values. Custom values from the user override values
+	// managed by the controlling operator.
+	chartConfigmapValues, err = mergeValuesConfigMaps(chartConfigmapValues, userConfigmapValues)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
```

Comment:
`mergeValuesConfigMaps` is where the complexity is. The rest of this is just getting values from the CR.